src/server.rs

```diff
@@ -402,3 +403,41 @@ pub extern "C" fn rustls_server_session_write_tls(
         rustls_result::Ok
     }
 }
+
+/// Write up to `len` characters of the SNI hostname into `buf`. If the
+/// handshake has not been performed yet or if the client does not support SNI,
+/// the returned length in out_n will be 0.
+/// https://docs.rs/rustls/0.19.0/rustls/struct.ServerSession.html#method.get_sni_hostname
+#[no_mangle]
+pub extern "C" fn rustls_server_session_sni_hostname_get(
+    session: *mut rustls_server_session,
+    buf: *mut c_char,
+    len: size_t,
+    out_n: *mut size_t,
+) -> rustls_result {
+    ffi_panic_boundary! {
+        let session: &mut ServerSession = try_ref_from_ptr!(session, &mut ServerSession);
```

Comment:
[`ServerSession::get_sni_hostname`](https://docs.rs/rustls/0.19.0/rustls/struct.ServerSession.html#method.get_sni_hostname) takes `&self`. Can `session` be an immutable reference?